drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1919,15 +1958,14 @@ static void eth0_iface_init(struct net_if *iface)
 		LOG_ERR("ETH PHY Initialization Error");
 		return;
 	}
-	/* PHY auto-negotiate link parameters */
-	result = phy_sam_gmac_auto_negotiate(&cfg->phy, &link_status);
-	if (result < 0) {
-		LOG_ERR("ETH PHY auto-negotiate sequence failed");
-		return;
-	}
 
-	/* Set up link parameters */
-	link_configure(cfg->regs, link_status);
+	/* Initialise monitor */
+	k_delayed_work_init(&dev_data->monitor_work, monitor_work_handler);
```

Comment:
So we're creating 1 work per Ethernet instance. I guess this is OK as low-end targets don't have multiple MAC-PHY pairs but in general we should consider 1 work per driver which handles auto-neg for all instances of a given IP. This will keep the load on the system minimal.

Reply:
> So we're creating 1 work per Ethernet instance. I guess this is OK as low-end targets don't have multiple MAC-PHY pairs but in general we should consider 1 work per driver which handles auto-neg for all instances of a given IP. This will keep the load on the system minimal.

"Work" in itself does not really create a lot of overhead- a workqueue does. As long as the polling process runs on the system workqueue, the overhead should be minimal and having one for all vs. one per instance would not make that much of a difference.